source/parse.h

```diff
@@ -6467,6 +6467,10 @@ class parser
 
         //  Next is optionally a requires clause
         if (curr() == "requires") {
+            if (!n->template_parameters) {
+                error("'requires' only valid for a template");
+                return {};
+            }
```

Comment:
It's also valid for templated entities (non-templates within templates).

Reply:
Could you provide an example?

Reply:
```Cpp2
t: <T> type = {
  f: () requires std::regular<T> = { }
}
```

Reply:
Also, I think the above won't handle the case of generic parameters that use wildcards instead of an explicit template parameter list? Such as

```
f: (x) requires std::regular<decltype(x)> = { }
```

Reply:
I do think there's value in diagnosing a `requires` on a non-templated type for _type_ declarations.

Rationale:
- writing a stray `requires` on a function already gives a nice Cpp1 diagnostic (e.g., MSVC says `error C7599: 't::f': a trailing requires clause is only allowed on a templated function`)
- writing a stray `requires` on a type currently lowers it to a naked `requires` at global scope before the type which leads to a harder-to-understand Cpp1 diagnostic (e.g., MSVC says `error C2059: syntax error: 'requires'`, which is correct and indicates the rough issue, but it's as specific)

So diagnosing the type case in Cpp2 delivers the primary diagnostic usability improvement, plus I think it's easy to code, I think it just needs a `n->is_type()` to the condition Filip already wrote

I'll modify the PR to try that out...